waiter/src/waiter/scheduler/cook.clj

```diff
@@ -477,11 +479,15 @@
 
   (state [_]
     {:service-id->failed-instances-transient-store @service-id->failed-instances-transient-store
-     :syncer (retrieve-syncer-state-fn)}))
+     :syncer (retrieve-syncer-state-fn)})
+
+  (validate-service [_ service-id]
+    (let [{:strs [run-as-user]} (service-id->service-description-fn service-id)]
+      (authz/check-user authorizer run-as-user service-id))))
 
 (s/defn ^:always-validate create-cook-scheduler
   "Returns a new CookScheduler with the provided configuration."
-  [{:keys [allowed-users backend-port home-path-prefix instance-priorities search-interval-days
+  [{:keys [allowed-users authorizer backend-port home-path-prefix instance-priorities search-interval-days
```

Comment:
Should we add schema validation for `authorizer` in the pre-condition?

Reply:
Done.